### ISSUE TITLE
Expected HTTP 101 response but was '504 Gateway Timeout'

### ISSUE THREAD

**Issue by cmoulliard** [fabric8io/fabric8-maven-plugin#1016]
When we do fabric8:deploy of a simple Spring Boot project against openshift 3.6 using minishift 1.4.1, we get randomly this error with f-m-p / 3.5.22

```
INFO] F8: Push successful
[WARNING] Exec Failure: HTTP 504, Status: 504 - Timeout: Too large resource version: 5899, current: 5898
java.net.ProtocolException: Expected HTTP 101 response but was '504 Gateway Timeout'
        at okhttp3.internal.ws.RealWebSocket.checkResponse(RealWebSocket.java:216)
        at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:183)
        at okhttp3.RealCall$AsyncCall.execute(RealCall.java:141)
        at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
[ERROR] F8: Failed to execute the build [Unable to build the image using the OpenShift build service]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 29.402 s
[INFO] Finished at: 2017-08-09T15:35:15+02:00
[INFO] Final Memory: 42M/469M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.22:build (default) on project fabric8-maven-sample-spring-boot: Failed to execute the build: Unable to build the image using the OpenShift build service: Timeout: Too large resource version: 5899, current: 5898 -> [Help 1]
[ERROR]
```

**Comment by jstrachan** [comment]
got a git repo with the project that failed?

**Comment by cmoulliard** [comment]
https://github.com/cmoulliard/issue-1016

**Comment by cmoulliard** [comment]
This error also appears intermittently when we issue 

```
mvn clean install fabric8:undeploy
mvn clean install fabric8:deploy
```

Error

```
[INFO] F8: Pushed 19/23 layers, 83% complete
[INFO] Current reconnect backoff is 1000 milliseconds (T0)
[INFO] F8: Pushed 20/23 layers, 87% complete
[INFO] F8: Pushed 21/23 layers, 91% complete
[INFO] F8: Pushed 22/23 layers, 96% complete
[INFO] F8: Pushed 23/23 layers, 100% complete
[INFO] F8: Push successful
[INFO] F8: Build fabric8-maven-sample-spring-boot-s2i-32 Complete
[INFO] Current reconnect backoff is 2000 milliseconds (T1)
[ERROR] Exception in reconnect
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@ebc280e rejected from java.util.concurrent.ScheduledThreadPoolExecutor@4bbeb7bd[Shutting down, pool size = 1, active threads = 1, queued tasks = 0, completed tasks = 2]
        at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2047)
        at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:823)
        at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:326)
        at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:533)
        at java.util.concurrent.Executors$DelegatedScheduledExecutorService.schedule(Executors.java:729)
        at io.fabric8.kubernetes.client.dsl.internal.WatchHTTPManager$3.run(WatchHTTPManager.java:210)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by utherp0** [comment]
I get the error consistently when trying to build any of the MicroServices in the new MSA Ref Arch example - https://access.redhat.com/articles/3155471

[WARNING] Exec Failure: HTTP 504, Status: 504 - Timeout: Too large resource version: 19202781, current: 19202780
java.net.ProtocolException: Expected HTTP 101 response but was '504 Gateway Timeout'
	at okhttp3.internal.ws.RealWebSocket.checkResponse(RealWebSocket.java:216)
	at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:183)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:141)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

**Comment by mojsha** [comment]
I can comment that we are also getting these with F-M-P 3.5.25 and OpenShift Origin 3.6.

**Comment by jstrachan** [comment]
@cmoulliard tried many times but can't reproduce the issue typing "mvn fabric8:deploy" using your project: https://github.com/cmoulliard/issue-1016 on minishift 1.4.1 with OS 3.6.0. Any ideas how to reproduce? Whats the exact CLI you type?

**Comment by Ladicek** [comment]
I'm seeing this very often as well, but I now realize that I only see it when using an actually _remote_ OpenShift instance. I wasn't able to reproduce it with an `oc cluster up` instance, but with a full remote OCP 3.6 cluster, it occurs very often / almost always. Executing just

```
git clone https://github.com/wildfly-swarm-openshiftio-boosters/wfswarm-rest-http.git
cd wfswarm-rest-http
mvn clean fabric8:deploy -Popenshift -DskipTests
```

(This project has FMP 3.5.22, but that should be OK.)

So naively, it sounds like a concurrency issue.

**Comment by Ladicek** [comment]
Just tried to reproduce with Minishift and there's a reasonably easy way to do that. Before running `mvn fabric8:deploy`, login to the Minishift VM (`minishift ssh`) and run

```bash
sudo tc qdisc add dev eth1 root handle 1:0 netem delay 150ms
```

(Run `ip a` to figure out which device has the `192.168.x.x` address, it's `eth1` in my case.)

**Comment by kevinearls** [comment]
I'm seeing this too, although I see it far more frequently when running a Jenkins build using a Jenkins instance running in OpenShift.  I'm also using minishift 1.4.1, and fpm 3.5.22 and 3.5.25.

**Comment by rhuss** [comment]
Unfortunately, I couldn't reproduce it (with @cmoulliard test repo and @Ladicek tip to add latency). 

Actually, it looks like a timeout on a WebSocket based watch for some results in order continue. We could be more conservative here and, when a watch breaks, do a check if the condition is fulfilled and if not start a new watch.

For me it's not really clear where this happens. Could you please tell me which actual goal is called when this happens (`deploy` is an aggregated goal calling resource+build+apply normally). A larger context around the error in the logs would be helpful, too.

**Comment by mojsha** [comment]
@rhuss It happens for the fabric8:build target. 

It happens (for me) regardless of which project or configuration I see, and on a random basis. 

I would concur with @kevinearls that it happens more frequently with a OpenShift-hosted Jenkins instance. 

There are two types of exceptions:

1. 
`ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.25-NETS:build (default-cli) on project api-mpa-gatekeeper: Failed to execute the build: Unable to build the image using the OpenShift build service: Timeout: Too large resource version: 30152684, current: 30152679 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.25-NETS:build (default-cli) on project api-mpa-gatekeeper: Failed to execute the build
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
       at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to execute the build
        at io.fabric8.maven.plugin.mojo.build.BuildMojo.buildAndTag(BuildMojo.java:245)
        at io.fabric8.maven.docker.BuildMojo.executeInternal(BuildMojo.java:44)
        at io.fabric8.maven.plugin.mojo.build.BuildMojo.executeInternal(BuildMojo.java:222)
        at io.fabric8.maven.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:223)
        at io.fabric8.maven.plugin.mojo.build.BuildMojo.execute(BuildMojo.java:193)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
        ... 20 more
Caused by: io.fabric8.maven.core.service.Fabric8ServiceException: Unable to build the image using the OpenShift build service
        at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.build(OpenshiftBuildService.java:97)
        at io.fabric8.maven.plugin.mojo.build.BuildMojo.buildAndTag(BuildMojo.java:242)
        ... 26 more
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Timeout: Too large resource version: 30152684, current: 30152679
        at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$2.onFailure(WatchConnectionManager.java:191)
        at okhttp3.internal.ws.RealWebSocket.failWebSocket(RealWebSocket.java:543)
        at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:185)
        at okhttp3.RealCall$AsyncCall.execute(RealCall.java:141)
        at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
`

This one breaks the process, i.e. no artifact is actually created.

2. `[INFO] F8: Build api-mpa-gatekeeper-s2i-3 status: Complete
[INFO] F8: Build api-mpa-gatekeeper-s2i-3 Complete
[INFO] Current reconnect backoff is 2000 milliseconds (T1)
[ERROR] Exception in reconnect
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@62ac9842 rejected from java.util.concurrent.ScheduledThreadPoolExecutor@27ae1a73[Shutting down, pool size = 1, active threads = 1, queued tasks = 0, completed tasks = 2]
        at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2047)
        at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:823)
        at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:326)
        at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:533)
        at java.util.concurrent.Executors$DelegatedScheduledExecutorService.schedule(Executors.java:729)
        at io.fabric8.kubernetes.client.dsl.internal.WatchHTTPManager$3.run(WatchHTTPManager.java:210)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
[INFO] F8: Found tag on ImageStream api-mpa-gatekeeper tag: sha256:c68241131d3cba2c5a1ec93368177e53f50370307d1f5ee5b276d5c3c7155a5e
[INFO] F8: ImageStream api-mpa-gatekeeper written to C:\projects\api-mpa-gatekeeper\target\api-mpa-gatekeeper-is.yml
`
This one is less frequent and is "benign" in that the image gets pushed up properly.

**Comment by kevinearls** [comment]
I'm seeing it both with the build and deploy targets.  It fails for me on the build target when run in Jenkins, but on the deploy target if I'm running at a command line.  The project is git@github.com:kevinearls/wildfly-swarm-opentracing-demo.git . I've attached logs for both failures.
[failOnBuild.txt](https://github.com/fabric8io/fabric8-maven-plugin/files/1244845/failOnBuild.txt)
[failOnDeploy.txt](https://github.com/fabric8io/fabric8-maven-plugin/files/1244846/failOnDeploy.txt)

**Comment by rhuss** [comment]
Sorry, I could reproduce it now (had fmp connected to the wrong cluster ;).

So it looks like an issue with the S2I build while we are waiting on the end of the S2I build via an watch on the build.

The error actually happens [here](https://github.com/rhuss/fabric8-maven-plugin/blob/ee16a94ce9babaec14a3b4f253c89e50302f55fb/core/src/main/java/io/fabric8/maven/core/service/openshift/OpenshiftBuildService.java#L283) where fmp is watching for the end of the build.

In the end, the okhttp is calling this [onFailure()](https://github.com/rhuss/kubernetes-client/blob/09bf3af945e5d6af40638d8f1105dd52d13600bc/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java#L163) method during dealing with the websocket. 

@iocanel any idea how we can make this more robust ? It looks like that this error happens before a successful open has been done. Interestingly the watch on the logs was successful.

I also wonder where this `Too large resource version: 14425, current: 14422` comes from. Seems to be sent by OpenShift when opening the watch.

**Comment by rhuss** [comment]
The watch is opened via `https://192.168.64.3:8443/apis/build.openshift.io/v1/namespaces/myproject/builds/fabric8-maven-sample-spring-boot-s2i-9?resourceVersion=14552&watch=true`

and the error is `504 - Timeout: Too large resource version: 14552, current: 14547` which indicates that there is some mismatch with the version of the build resource on the OpenShift side ?

**Comment by rhuss** [comment]
Next step: When looking at the watch code it works like that the resourceVersion is fetched from the BuildList's metadata during construction of the watch manager objects, but when the actual wtch is created this number has been already increased.

Actually, when I re-fetch the BuildList in the debugger, the revision number is constantly increasing (i.e. is larger after each call to `BuildOperationsImpl.list()`)

I don't know why this is happening, but this cries out for a race condition like here. I.e. fetching the version number first and then creating the watch in a second operation is not atomic.

And finally, I really wonder why we are watching on the whole BuildList (which contains all Builds) and not on the individual Build object that we want to monitor. @iocanel any idea why the `WatchConnectionManager` in kube-client is operating always on a list of objects and not on an individual object ?

The interesting thing is, that the resourceVersion (14552 this is the one which is increasing) from the list() operation is larger than mentioned as 'current' in the error message (14547).

Sorry, really confused here.

**Comment by jstrachan** [comment]
I managed to reproduce it too:
```
[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.22:build (default) on project flights: Failed to execute the build: Unable to build the image using the OpenShift build service: Timeout: Too large resource version: 19393279, current: 19393278 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.22:build (default) on project flights: Failed to execute the build
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:213)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
	at org.apache.maven.lifecycle.internal.MojoExecutor.executeForkedExecutions(MojoExecutor.java:353)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:198)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to execute the build
	at io.fabric8.maven.plugin.mojo.build.BuildMojo.buildAndTag(BuildMojo.java:245)
	at io.fabric8.maven.docker.BuildMojo.executeInternal(BuildMojo.java:44)
	at io.fabric8.maven.plugin.mojo.build.BuildMojo.executeInternal(BuildMojo.java:222)
	at io.fabric8.maven.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:223)
	at io.fabric8.maven.plugin.mojo.build.BuildMojo.execute(BuildMojo.java:193)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 24 more
Caused by: io.fabric8.maven.core.service.Fabric8ServiceException: Unable to build the image using the OpenShift build service
	at io.fabric8.maven.core.service.openshift.OpenshiftBuildService.build(OpenshiftBuildService.java:97)
	at io.fabric8.maven.plugin.mojo.build.BuildMojo.buildAndTag(BuildMojo.java:242)
	... 30 more
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Timeout: Too large resource version: 19393279, current: 19393278
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$2.onFailure(WatchConnectionManager.java:191)
	at okhttp3.internal.ws.RealWebSocket.failWebSocket(RealWebSocket.java:543)
	at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:185)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:141)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```
@rhuss yeah - it sounds like we need to do a watch without the revision number on the named build. I'm not even sure why we should include the revision number in the watch - as the version changes anyways whenever its updated?

**Comment by jstrachan** [comment]
I wonder if we temporarily removed the `resourceVersion` from the watch it might help things?
e.g. `https://192.168.64.3:8443/apis/build.openshift.io/v1/namespaces/myproject/builds/fabric8-maven-sample-spring-boot-s2i-9?watch=true`

**Comment by rhuss** [comment]
It's an internal part of kubernetes-client (we only call `watch()` from f-m-p), but this might be a solution. Still no idea, why OpenShift consists a much older resourceVersion as current then the one returned by list()

@iocanel any idea ? Should I open an issue on kubernetes-client ?

**Comment by mojsha** [comment]
Any ideas on how to move this forward? It's causing us alot of issues as most of our builds are now failing.

**Comment by rhuss** [comment]
The proper solution is imo a fix of kubernetes client, let me open an issue over there.

I'm not sure why the watch is attached to the resourceVersion tbh. @iocanel is off this week, but this should be tackled as soon as possible.

**Comment by mojsha** [comment]
Ok, in the meanwhile I tried "downgrading" f-m-p version and it started to work normally for me at 3.5.19 so perhaps there was some kind of change there? Not sure.

**Comment by siamaksade** [comment]
I get this problem also with 3.5.19

**Comment by nicolaferraro** [comment]
I've just created a [PR in Kubernetes-client](https://github.com/fabric8io/kubernetes-client/pull/855) that fixes this problem.

@jstrachan, @rhuss

**Comment by jstrachan** [comment]
@nicolaferraro the new [kubernetes-client 2.6.3](http://central.maven.org/maven2/io/fabric8/kubernetes-client/2.6.3/) is now in maven central - I wonder if using that fixes this issue?

**Comment by nicolaferraro** [comment]
Yes, this issue is fixed.
@jstrachan it's better to wait for #1040 before triggering a release.

**Comment by Ladicek** [comment]
For reference (since this was closed without a linked PR or something), the last FMP release as of now is 3.5.26, so the expectation is that this will be fixed in 3.5.27 (or something like this).

**Comment by mojsha** [comment]
This is causing us major headache - if I understand correctly this has been fixed in the subpackage, but not yet merged into F-M-P? When do we think that will happen?

**Comment by jstrachan** [comment]
@mojsha we should have a new release pretty soon

**Comment by jstrachan** [comment]
@mojsha 3.5.28 should appear in maven central within 20-30 mins

**Comment by mojsha** [comment]
@jstrachan Thanks alot! Really appreciate it.
